### ISSUE TITLE
Getting the 'closest' is broken sometimes

### ISSUE THREAD

**Issue by fm94** [CN-TU/pysdo#2]
When running the following code:
```
import numpy as np
from pysdo import SDO

SEED=2020
coreset_size=5

data = np.load('Dataset_0.npy')
model = SDO(k = coreset_size, random_state=SEED)
model.fit(data)
```
I get the following error:
```
--> 174                 closest = model.query(X[i:(i+self.chunksize)], return_distance=False, k=self.x).flatten()
    175                 #dist = distance.cdist(X[i:(i+self.chunksize)], observers, self.metric)
    176                 #dist_sorted = np.argsort(dist, axis=1)

sklearn/neighbors/binary_tree.pxi in sklearn.neighbors.kd_tree.BinaryTree.query()

ValueError: k must be less than or equal to the number of training points
```
Apparently, the issue is in this line
https://github.com/CN-TU/pysdo/blob/15bc50e6b91cd3dd08cdbc9321363df3493fe31c/pysdo/__init__.py#L161

And the old (commented) code works:
```
#dist = distance.cdist(X[i:(i+self.chunksize)], observers, self.metric)
#dist_sorted = np.argsort(dist, axis=1)
#closest = dist_sorted[:,0:self.x].flatten()
```

Data to reproduce the error:
[Dataset_0.zip](https://github.com/CN-TU/pysdo/files/4724053/Dataset_0.zip)

**Comment by alexhartl** [comment]
pysdo defaults to `x=6`, so with `k=coreset_size=5` you tell it that you want to use the 6 nearest observers, even though there are only 5 observers in total. Does the error persist when increasing the `k`?

**Comment by fm94** [comment]
Ah, problem solved. 

Interestingly, despite the fact that the old code is slower but it' was more robust due to numpy not checking the length.

I created a pull request to raise an error if a user enters x>k by mistake.

**Comment by alexhartl** [comment]
Fixed in #3.
